alternator/executor.cc

```diff
@@ -3383,6 +3388,9 @@ update_item_operation::update_item_operation(service::storage_proxy& proxy, rjso
         if (!_attribute_updates->IsObject()) {
             throw api_error::validation("AttributeUpdates must be an object");
         }
+        for (auto it = std::as_const(*_attribute_updates).MemberBegin(); it != std::as_const(*_attribute_updates).MemberEnd(); ++it) {
+            validate_attr_name_length(CONTEXT_NAME_MAIN_TABLE, it->name.GetStringLength(), false);
```

Comment:
I think the use of CONTEXT_NAME_MAIN_TABLE here is irrelevant (in ExpressionAttributeNames below it's even worse, it can be wrong).
The "context" of the validation here is "AttributeUpdates", not "main table".

This strengthens my original belief that the "context" of the validation can be many different strings - sometimes you're checking the KeySchema of a GSI or main table - but in other places you're checking AttributeUpdates or ExpressionAttributeNames, and those strings are useful contexts to print in the message - not "Main Table".